www/src/components/app/upgrade/Upgrade.tsx

```diff
@@ -6,36 +6,50 @@ import {
   ListBoxItem,
   PageTitle,
   Select,
+  Switch,
   Tooltip,
   WrapWithIf,
 } from '@pluralsh/design-system'
-import { Div, Flex, P, Switch } from 'honorable'
+import { Div, Flex, P } from 'honorable'
 import { isEmpty } from 'lodash'
-import { useEffect, useMemo, useState } from 'react'
+import { useCallback, useMemo } from 'react'
 import { useMutation } from '@apollo/client'
 
 import { UPDATE_INSTALLATION } from '../../repository/queries'
 import { useAppContext } from '../../../contexts/AppContext'
 import { GqlError } from '../../utils/Alert'
 import { AppHeaderActions } from '../AppHeaderActions'
+import { useUpdateState } from '../../../hooks/useUpdateState'
 
 export function Upgrade() {
   const { installation, upgradeChannels = [] } = useAppContext()
-  const [autoUpgrade, setAutoUpgrade] = useState(
-    installation?.autoUpgrade || false
-  )
-  const [trackTag, setTrackTag] = useState(installation?.trackTag)
+
+  const {
+    state: formState,
+    update: updateFormState,
+    hasUpdates,
+  } = useUpdateState({
+    autoUpgrade: installation?.autoUpgrade,
+    trackTag: installation?.trackTag,
+  })
+  const { autoUpgrade, trackTag } = formState
   const [mutation, { loading, error }] = useMutation(UPDATE_INSTALLATION, {
-    variables: { id: installation?.id, attributes: { trackTag, autoUpgrade } },
+    variables: {
+      id: installation?.id,
+      attributes: { trackTag, autoUpgrade },
+    },
   })
   const hasUpgradeChannels = useMemo(
     () => !isEmpty(upgradeChannels),
```

Comment:
this definitely shouldn't be reading as empty most of the time, apps will at least always have a latest channel

Reply:
oh i see we needed to add the `upgradeChannels` field to the gql query as well